File: 169-majority-element/majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {

        Dictionary<int, int> map = new Dictionary<int, int>();
        for(int i=0;i<nums.Length;i++){
            if(map.ContainsKey(nums[i])){
                map[nums[i]]++;
            }else{
                map.Add(nums[i],1);
            }
        }

        int max = 0;
        int ans = 0;
        foreach(var item in map){
            int val = item.Value;
            if(max <val )
            {
                max = val;
                ans = item.Key;
            }
        }

        return ans;

    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
            unordered_map<int,int> map ;

            for(int i = 0;i<nums.size();i++){
                    if(map.find(nums[i])!=map.end()){
                        map[nums[i]]++;
                    }else{
                        map[nums[i]] = 1;
                    }
            }
            int maxval = 0;
            int ans = 0;
            for(auto &it : map){
                if(maxval < it.second){
                    maxval = it.second;
                    ans = it.first;
                }
            }

            return ans;

    }
};